shell/shell.go

```diff
@@ -0,0 +1,104 @@
+package shell
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"sync"
+
+	"github.com/google/shlex"
+)
+
+// Default Inteface to use if none provided.
+var Default = "bash"
+
+// Interface to the shell.
+type Interface interface {
+	// Name of the shell interface.
+	Name() string
+
+	// Put in a exec command line and get a Cmd.
+	Command(line string) (*exec.Cmd, error)
+}
+
+var (
+	shellsLock sync.Mutex
+	shells     = make(map[string]Interface)
+)
+
+func init() {
+	Register(&Raw{})
+	Register(&Bash{})
+}
+
+// Register a new shell interface.
+func Register(i Interface) {
```

Comment:
I think we can do without the Register function and the invocations in init. We'll have a small, fixed number of shells (bash, raw, cmd... maybe powershell?), so we can just enter them into the table in the var declaration.